common/wireaddr.h

```diff
@@ -120,8 +123,13 @@ struct wireaddr_internal {
 		struct wireaddr wireaddr;
 		/* ADDR_INTERNAL_ALLPROTO */
 		u16 port;
-		/* ADDR_INTERNAL_AUTOTOR */
-		struct wireaddr torservice;
+		/* ADDR_INTERNAL_AUTOTOR
+		 * ADDR_INTERNAL_STATICTOR */
+		struct torservice {
+			struct wireaddr torservice_address;
+			u16 port;
+			char blob[TOR_V3_BLOBLEN + 1];
```

Comment:
u8 blob, instead of char?  And I'm not sure if the +1 is needed...

Reply:
done; I kept the + 1 for now ...